src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_GROW_INDEX = 2;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size + 1 > threshold) {
+            resizeTable();
+        }
+        Node<K, V> current = new Node<>(key, value, null);
+        int index;
+        if (key == null) {
+            index = 0;
+        } else {
+            index = Math.abs(getHash(key) % table.length);
+        }
+        if (table[index] == null) {
+            table[index] = current;
+            size++;
+        } else {
+            Node<K, V> buffer = table[index];
+            while (buffer != null) {
+                if (Objects.equals(buffer.key, current.key)) {
+                    buffer.value = current.value;
+                    break;
+                } else if (buffer.next == null) {
+                    buffer.next = current;
+                    size++;
+                    break;
+                } else {
+                    buffer = buffer.next;
+                }
+            }
+        }
+
```

Comment:
In my opinion, it will be better if u will split the logic of this method into 2 different methods (for example putInFull(if this bucket already contains elements) and putInEmpty(if this bucket is empty). It may improve the readability of your code. 
**You can ignore this suggestion it is just my opinion!**

Reply:
it's  change nothing. Code will be same